core/templates/dev/head/services/contextual/UrlServiceSpec.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the BackgroundMaskService.
+ */
+
+describe('Url Service', function() {
+  var UrlService = null;
+  var parameterList = '?parent=parent1&parent=parent2';
+  var sampleHash = 'sampleHash';
+  var window = {
+    href: 'http://sample.com/embed' + parameterList,
+    pathname: 'sample.com/embed',
```

Comment:
Instead of duplicating, maybe declare `var pathname`  above and construct the href value as `'http://' + pathname + queryString`.

Reply:
Done